### PR TITLE
Pass insecure parameter to NewTokenProvider

### PR DESCRIPTION
**Pull request by TaoZou1**

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -16,6 +17,7 @@ import (
 //TODO replace to yaml
 const (
 	nsxOperatorDefaultConf = "/etc/nsx-operator/nsxop.ini"
+	vcHostCACertPath       = "/etc/vmware/wcp/tls/vmca.pem"
```
Comment (dantingl) [inline pkg/config/config.go:20]: I guess this is the default path, is there a config option for the path?

Comment (TaoZou1) [reply pkg/config/config.go:20]: no, ncp uses it directly.

File: pkg/config/config.go [hunk 2]
```diff
@@ -61,6 +63,7 @@ type VCConfig struct {
 	VCEndPoint string `ini:"vc_endpoint"`
 	SsoDomain  string `ini:"sso_domain"`
 	HttpsPort  int    `ini:"https_port"`
+	VCInsecure bool   `ini:"insecure"`
```
Comment (dantingl) [inline pkg/config/config.go:66]: There is no VC `insecure` in configmap

File: pkg/config/config.go (hunk 1 above)
Comment (TaoZou1) [reply pkg/config/config.go:20]: no

File: pkg/config/config.go (hunk 2 above)
Comment (TaoZou1) [reply pkg/config/config.go]: use insecure in NsxConfig

File: pkg/config/config.go [hunk 3]
```diff
@@ -150,8 +152,16 @@ func (operatorConfig *NSXOperatorConfig) GetTokenProvider() auth.TokenProvider {
 }
 
 func (operatorConfig *NSXOperatorConfig) createTokenProvider() error {
+	log.Info("try to load VC host CA")
+	vcCaCert, err := ioutil.ReadFile(vcHostCACertPath)
+	// If operatorConfig.VCInsecure is false, tls will the CA to verify the server
+	// certificate. If loading CA failed, tls will use the CA on local host
+	if err != nil {
+		log.Info("fail to load CA cert from file", "error", err)
+	}
```
Comment (dantingl) [inline pkg/config/config.go:161]: Do we load VC cert only when insecure = false?

File: pkg/config/config.go [hunk 4]
```diff
@@ -150,8 +152,16 @@ func (operatorConfig *NSXOperatorConfig) GetTokenProvider() auth.TokenProvider {
 }
 
 func (operatorConfig *NSXOperatorConfig) createTokenProvider() error {
+	log.Info("try to load VC host CA")
```
Comment (dantingl) [inline pkg/config/config.go:155]: debug log level should be enough

Comment (TaoZou1) [reply pkg/config/config.go:155]: done

File: pkg/config/config.go (hunk 3 above)
Comment (TaoZou1) [reply pkg/config/config.go:161]: yes. 
updated.

**Review by dantingl — APPROVED** [review]

**Pull request merged**